eng/templates/jobs/build-from-source.yml

```diff
@@ -0,0 +1,138 @@
+  - name: 'Centos71_Portable'
+    include: 'always'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-3e800f1-20190501005343'
+  
+  # If managedOnly is false, run a macOS portable build
+  - name: 'MacOS_Portable'
+    jobProperties:
+      pool:
+        name: Hosted macOS
+
+  # Non-portable platforms
+  - name: 'Centos71'
+    targetRID: 'centos.7-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-3e800f1-20190501005343'
+
+  - name: 'Centos8'
+    targetRID: 'centos.8-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-8-daa5116-20200325130212'
+
+  - name: 'Debian9'
+    targetRID: 'debian.9-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:debian-stretch-d61254f-20190807161114'
+
+  - name: 'Fedora30'
+    targetRID: 'fedora.30-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-30-38e0f29-20191126135223'
+
+  - name: 'Fedora32'
+    targetRID: 'fedora.32-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-32-20200512010618-163ed2a'
+
+  - name: 'MacOS'
+    targetRID: 'osx.10.14-x64'
+    jobProperties:
+      pool:
+        name: Hosted macOS
+
+  - name: 'Ubuntu1804'
+    targetRID: 'ubuntu.18.04-x64'
+    jobProperties:
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-f90bc20-20180320154721'
+
+jobs:
+- job: Build_From_Source_Main
+  displayName: Build From Source Main
+  pool: server
+  dependsOn:
+  - ${{ each platform in parameters.platforms }}:
+    - ${{ if or(eq(platform.include, 'always'), eq(parameters.managedOnly, false)) }}:
+      - ${{ parameters.jobNamePrefix }}_${{ platform.name }}  
+
+- ${{ each platform in parameters.platforms }}:
+  - ${{ if or(eq(platform.include, 'always'), eq(parameters.managedOnly, false)) }}:
+    - job: ${{ parameters.jobNamePrefix }}_${{ platform.name }}
+      displayName: Build From Source (${{ platform.name }})
+      ${{ each property in platform.jobProperties }}:
+        ${{ property.key }}: ${{ property.value }}
+      workspace:
+        clean: all
+
+      variables:
+      - name: _BuildConfig
+        value: Release
+      - name: _InternalBuildArgs
+        value: ''
+      - name: _NonPortableBuildArgs
+        value: ''        
+      - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+        - name: _InternalBuildArgs
+          value: >-
+            /p:DotNetPublishUsingPipelines=true
+            /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
+      - ${{ if ne(platform.targetRID, '') }}:
+        - name: _NonPortableBuildArgs
+          value: >-
+            /p:ArcadeBuildNonPortable=true
+            /p:TargetRid=${{ platform.targetRID }}
+
+      steps:
+      - script: |
+          set -x
+          df -h
+          ./build.sh --ci \
```

Comment:
Something I realized we'll definitely need is a way to change the root script, for repos that aren't onboarded to an Arcade entrypoint like NuGet.Client. (Maybe this is only NuGet.Client? Not sure.) Then they can change it to an `eng/source-build.sh` or whatever, and we don't have to worry about letting people also configure the args in this template.

(Defaulting to `./build.sh` seems good though.)

Reply:
Thanks.  I added a parameter called `buildScript` and initialized it to `./build.sh`.